### PR TITLE
Cu player portal v2

### PR DESCRIPTION
**Pull request by kellan-simiotics**
## Changes

Adding inventory page for Crypto Unicorns game.

## How to test these changes?

Tested mostly on Mumbai.

### CODE REVIEW

**Comment by kellan-simiotics** [comment]
@zomglings Ready for review.

File: apps/play/package.json [hunk 1]
```diff
@@ -8,9 +8,11 @@
     "start": "next start",
     "lint": "eslint ./ --ext js,jsx,ts,tsx --fix",
     "pretty": "prettier --write \"./**/*.{js,jsx,json,ts,tsx}\"",
-    "generate_classes": "typechain --target=web3-v1 --out-dir ../../types/contracts '../../build/contracts/*.json'"
+    "generate_classes": "typechain --target=web3-v1 --out-dir ../../types/contracts '../../build/contracts/*.json'",
+    "generate_classes_cu": "typechain --target=web3-v1 --out-dir games/cu 'games/cu/*.json'"
   },
   "dependencies": {
+    "ethereum-multicall": "^2.14.1",
```
Comment (zomglings) [inline apps/play/package.json:15]: We are not currently using this dependency. Remove please.

File: packages/moonstream-components/src/core/providers/AnalyticsProvider/index.js [hunk 2]
```diff
@@ -108,10 +108,10 @@ const AnalyticsProvider = ({ children, mixpanelToken }) => {
     }
     const urlForUnmount = router.nextRouter.pathname;
     const closeListener = () => {
-      mixpanel?.track(MIXPANEL_EVENTS.PAGEVIEW_DURATION, {
-        url: urlForUnmount,
-        isBeforeUnload: true,
-      });
+      // mixpanel?.track(MIXPANEL_EVENTS.PAGEVIEW_DURATION, {
```
Comment (zomglings) [inline packages/moonstream-components/src/core/providers/AnalyticsProvider/index.js:111]: Uncomment for now - fix in separate PR but let's not change analytics behavior like this in this release.

File: apps/play/pages/games/cryptoUnicorns.tsx [hunk 3]
```diff
@@ -12,85 +12,857 @@ import {
...
   const [needAllowanceUNIM, setNeedAllowanceUNIM] = React.useState(false);
   const [rbwToStash, setRBWToStash] = React.useState("");
   const [unimToStash, setUNIMToStash] = React.useState("");
+  const [terminusAddress, setTerminusAddress] = React.useState("");
+  const [UNIMAddress, setUNIMAddress] = React.useState("");
+  const [RBWAddress, setRBWAddress] = React.useState("");
+  const [unicornsAddress, setUnicornsAddress] = React.useState("");
+  const [landsAddress, setLandsAddress] = React.useState("");
+
+  const [lootboxBalances, setLootboxBalances] = React.useState({
```
Comment (zomglings) [inline apps/play/pages/games/cryptoUnicorns.tsx:544]: Rename this from `lootboxBalances` to `terminusBalances`. Doesn't have to happen on this PR. Refactor later.

File: apps/play/pages/games/cryptoUnicorns.tsx [hunk 4]
```diff
@@ -12,85 +12,857 @@ import {
...
+  mumbai: "0x39858b1A4e48CfFB1019F0A15ff54899213B3f8b",
+  polygon: "0xdC0479CC5BbA033B3e7De9F178607150B3AbCe1f",
+  ethereum: "0x0000000000000000000000000000000000000000",
+  localhost: "0x0000000000000000000000000000000000000000",
+};
+
+const LandsAddresses: { [key in supportedChains]: string } = {
+  mumbai: "0x230E4e85d4549343A460F5dE0a7035130F62d74C",
+  polygon: "0xA2a13cE1824F3916fC84C65e559391fc6674e6e8",
+  ethereum: "0x0000000000000000000000000000000000000000",
```
Comment (zomglings) [inline apps/play/pages/games/cryptoUnicorns.tsx:91]: We can get rid of `ethereum` and `localhost` keys. Not so useful and add a lot of noise to the code.

File: apps/play/pages/games/cryptoUnicorns.tsx [hunk 5]
```diff
@@ -12,85 +12,857 @@ import {
...
+      }
+
+      setUnicorns(unicornsInventory);
+    },
+    {
+      ...hookCommon,
+    }
+  );
+
+  const fetchLands = useQuery(
```
Comment (zomglings) [inline apps/play/pages/games/cryptoUnicorns.tsx:759]: Add support for Shadowcorns - can be separate PR.

File: apps/play/pages/games/cryptoUnicorns.tsx [hunk 6]
```diff
@@ -12,85 +12,857 @@ import {
...
+        tokenIDs.forEach((tokenID, index) => {
+          landsInventory.push({
+            tokenID,
+            tokenURI: tokenURIs[index],
+            imageURI: imageURIs[index],
+            metadata: tokenMetadata[index],
+          });
+        });
+
+        // console.log(
```
Comment (zomglings) [inline apps/play/pages/games/cryptoUnicorns.tsx:827]: Remove comment.

File: apps/play/pages/games/cryptoUnicorns.tsx [hunk 7]
```diff
@@ -12,85 +12,857 @@ import {
...
+        let numLands: number = 0;
+        try {
+          numLands = parseInt(numLandsRaw, 10);
+        } catch (e) {
+          console.error(
+            `Error: Could not parse number of owned lands as an integer: ${numLandsRaw}`
+          );
+        }
+
+        console.log(`${prefix} numLands=${numLands}`);
```
Comment (zomglings) [inline apps/play/pages/games/cryptoUnicorns.tsx:794]: I think we can remove these `console.log`s.

**Review by zomglings — APPROVED** [review]
Left a few comments. Small changes. Approving. Please make some modifications before merging.

**Comment by zomglings** [comment]
Open issues to fix in next version: https://github.com/bugout-dev/engine/issues/186
